### ISSUE TITLE
Procs evaluated during initialization causing app to crash

### ISSUE THREAD

**Issue by Amnesthesia** [magnusvk/counter_culture#347]
When using scopes with a proc, like so:

```
counter_culture :account,
                  column_name: proc { |model|
                    model.role.slug == 'admin' ? 'admin_count' : 'user_count'
                  },
                  column_names: -> {{
                    UserAccount.admins => :admin_count,
                    UserAccount.users => :user_count,
                  }}
```

This is being evaluated here: https://github.com/magnusvk/counter_culture/blob/master/lib/counter_culture/extensions.rb#L53 

When CI (or anything really) then runs `rails db:schema:load` (or even just `rails db:schema:load_config`) this crashes because the database isn't set up yet and it's trying to evaluate the scopes, failing with:

```
Caused by:
PG::UndefinedTable: ERROR:  relation "user_accounts" does not exist
LINE 8:  WHERE a.attrelid = '"user_accounts"'::regclass
```

**Comment by AntoineBecquet** [comment]
A similar problem occurs when the proc use a scope defined after it in the model class.

Like in the readme, this works :

```ruby
class Subscription < ApplicationRecord
  belongs_to :project

  scope :contributing, -> { paid.or(initiated.reservations) }

  counter_culture :project,
                  column_name:  proc { |subscription| :amount_contributed if subscription.contributing? },
                  column_names: proc { {Subscription.contributing => :amount_contributed} },
                  delta_column: :amount
end
```
but this raise an error when the class is loaded :
```ruby
class Subscription < ApplicationRecord
  belongs_to :project

  counter_culture :project,
                  column_name:  proc { |subscription| :amount_contributed if subscription.contributing? },
                  column_names: proc { {Subscription.contributing => :amount_contributed} },
                  delta_column: :amount

  scope :contributing, -> { paid.or(initiated.reservations) }
end
```

`undefined method 'contributing' for #<Class:0x00007ff214788670> (NoMethodError)` comming as above from https://github.com/magnusvk/counter_culture/blob/master/lib/counter_culture/extensions.rb#L53.

As stated in the readme, we want to use procs to avoid calling their content on startup, so calling them inside `counter_culture` method breaks their purpose.

**Comment by magnusvk** [comment]
This should be fixed in gem version 3.3.0. Sorry for the delay.
